tests/unit/output/test_text_output.py

```diff
@@ -138,39 +138,74 @@ def test_leak_message(result_input, snapshot, show_secrets, verbose):
     snapshot.assert_match(output)
 
 
-def assert_policies_displayed(output, policy_breaks):
+def assert_policies_displayed(output, verbose, ignore_known_secrets, policy_breaks):
+    nb_new_secrets = 0
     for policy_break in policy_breaks:
-        if policy_break.known_secret:
-            assert f"Known secret: {policy_break.break_type}" in output
-        else:
+        # All secrets are displayed no matter if they're known or not
+        if not ignore_known_secrets:
+            nb_new_secrets += 1
             assert f"Secret detected: {policy_break.break_type}" in output
+        else:
+            # In that case known secrets are still displayed with a dedicated message
```

Comment:
(nitpick) I would find it more readable to have the comment inside the blocks, like this:
```
if verbose:
    # comment about the verbose block
else:
    # comment about the !verbose block
```